llama_index/llms/nvidia_triton.py

```diff
@@ -0,0 +1,581 @@
+        invocation_params["prompt"] = [[prompt]]
+        model_params = self._identifying_params
+        model_params.update(kwargs)
+        request_id = str(random.randint(1, 9999999))  # nosec
+
+        if self.triton_load_model_call:
+            client.load_model(model_params["model_name"])
+
+        result_queue = self._client.request_streaming(
+            model_params["model_name"], request_id, **invocation_params
+        )
+
+        response = ""
+        for token in result_queue:
+            if isinstance(token, InferenceServerException):
+                self._client.stop_stream(model_params["model_name"], request_id)
+                raise token
+            response = response + token
+
+        return CompletionResponse(
+            text=response,
+        )
+
+    def _stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
+        """Request a streaming inference session."""
+        client = self._get_client()
+
+        def gen() -> CompletionResponseGen:
+            invocation_params = self._get_model_default_parameters
+            invocation_params.update(kwargs)
+            invocation_params["prompt"] = [[prompt]]
+            model_params = self._identifying_params
+            model_params.update(kwargs)
+            request_id = str(random.randint(1, 9999999))  # nosec
+
+            result_queue = client.request_streaming(
+                model_params["model_name"], request_id, **invocation_params
+            )
+
+            text = ""
+            for token in result_queue:
+                text = text + token
+                yield CompletionResponse(
+                    text=text,
+                )
+
+        return gen()
+
+    @llm_chat_callback()
+    def chat(self, messages: Sequence[ChatMessage], **kwargs: Any) -> ChatResponse:
+        chat_fn = completion_to_chat_decorator(self._complete)
+        return chat_fn(messages, **kwargs)
+
+    def stream_chat(
+        self, messages: Sequence[ChatMessage], **kwargs: Any
+    ) -> ChatResponseGen:
+        raise NotImplementedError
+
+    def complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
+        return self._complete(prompt, **kwargs)
+
+    def stream_complete(self, prompt: str, **kwargs: Any) -> CompletionResponseGen:
+        raise NotImplementedError
+
+    async def achat(
+        self, messages: Sequence[ChatMessage], **kwargs: Any
+    ) -> ChatResponse:
+        raise NotImplementedError
+
+    async def acomplete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
+        raise NotImplementedError
+
+    async def astream_chat(
+        self, messages: Sequence[ChatMessage], **kwargs: Any
+    ) -> ChatResponseAsyncGen:
+        raise NotImplementedError
+
+    async def astream_complete(
+        self, prompt: str, **kwargs: Any
+    ) -> CompletionResponseAsyncGen:
+        raise NotImplementedError
+
+
+class StreamingResponseGenerator(queue.Queue[Optional[str]]):
```

Comment:
Just for the sake of organization, can we move this and the client into a sibling file like `nvidia_triton_utils.py` ?